gc/base/SweepPoolManagerAddressOrderedListBase.cpp

```diff
@@ -383,9 +412,16 @@ MM_SweepPoolManagerAddressOrderedListBase::flushFinalChunk(
 					sweepState->_connectPreviousFreeEntrySize,
 					sweepState->_connectPreviousChunk->trailingFreeCandidate);
 
+			if (memoryPool->doesNeedClearCardForConnectFreeEntry()) {
+				memoryPool->cleanCardsForFreeEntry(envModron, (void*)sweepState->_connectPreviousFreeEntry, ((uint8_t*)sweepState->_connectPreviousFreeEntry + sweepState->_connectPreviousFreeEntrySize), false);
+			}
+
 			sweepState->_connectPreviousPreviousFreeEntry = sweepState->_connectPreviousFreeEntry;
 			sweepState->_connectPreviousFreeEntry = (MM_HeapLinkedFreeHeader *)sweepState->_connectPreviousChunk->trailingFreeCandidate;
 			sweepState->_connectPreviousFreeEntrySize = sweepState->_connectPreviousChunk->trailingFreeCandidateSize;
+			if ((_extensions->isVLHGC()) && (envModron->_cycleState->_noCompactionAfterSweep)) {
```

Comment:
could this be moved up a bit so it goes into (what I proposed to be) connectOuterMemoryPostProcess?